### PR TITLE
[CXF-8378]:Fix NoClassDefFoundError: org/apache/cxf/common/util/Refle…

### PR DESCRIPTION
**Pull request by jimma**
…ctionUtil

### CODE REVIEW

File: rt/transports/http/src/main/java/org/apache/cxf/transport/http/ReferencingAuthenticator.java [hunk 1]
```diff
@@ -122,15 +125,41 @@ PasswordAuthentication tryWith(Authenticator a) throws Exception {
         if (a == null) {
             return null;
         }
-        for (final Field f : ReflectionUtil.getDeclaredFields(Authenticator.class)) {
+        Field[] fileds = null;
```
Comment (andrei-ivanov) [inline rt/transports/http/src/main/java/org/apache/cxf/transport/http/ReferencingAuthenticator.java:128]: typo: `fileds`

Comment (jimma) [reply rt/transports/http/src/main/java/org/apache/cxf/transport/http/ReferencingAuthenticator.java]: Fixed. Thanks @andrei-ivanov .

**Review by andymc12 — APPROVED** [review]
LGTM - my only comment would be that you could check for the security manager in a static boolean and save repeated calls to `System.getSecurityManager() == null`.  Thanks!

**Comment by jimma** [comment]
> LGTM - my only comment would be that you could check for the security manager in a static boolean and save repeated calls to `System.getSecurityManager() == null`. Thanks!

Thanks for review @andymc12 .  Changed this as you suggested.

**Pull request merged**